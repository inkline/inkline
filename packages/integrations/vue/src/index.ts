export * from './composables';
export * from './constants';
export * from './plugins';
export * from './plugin';
export * from './types';
