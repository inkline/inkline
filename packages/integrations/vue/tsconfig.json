{
    "extends": "../../tooling/typescript-config/tsconfig.json",
    "compilerOptions": {
        "types": [
            "vite/client",
            "vitest/globals"
        ]
    },
    "include": [
        "src/**/*.ts",
        "tsup.config.ts",
        "vitest.config.ts"
    ]
}
