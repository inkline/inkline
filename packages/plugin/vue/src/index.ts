export * from './plugin';
export * from './symbols';
