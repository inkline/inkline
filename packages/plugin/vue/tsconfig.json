{
    "extends": "../../tooling/typescript-config/tsconfig.json",
    "compilerOptions": {
        "paths": {
            "@inkline/types": ["../../types/src/*"]
        },
        "types": [
            "vite/client",
            "vitest/globals"
        ]
    },
    "include": [
        "src/**/*.ts",
        "tsup.config.ts",
        "vite.config.ts"
    ]
}
