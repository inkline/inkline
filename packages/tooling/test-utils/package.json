{
    "name": "@inkline/test-utils",
    "version": "1.0.0",
    "type": "module",
    "scripts": {
        "dev": "devtools tsup watch",
        "build": "devtools tsup build",
        "lint": "devtools eslint"
    },
    "dependencies": {
        "@inkline/vue": "workspace:*"
    },
    "devDependencies": {
        "@inkline/devtools": "workspace:*",
        "@inkline/eslint-config": "workspace:*",
        "@inkline/types": "workspace:*",
        "@types/node": "catalog:",
        "@types/shelljs": "catalog:",
        "eslint": "catalog:",
        "tsx": "catalog:",
        "tsup": "catalog:",
        "typescript": "catalog:",
        "type-fest": "catalog:",
        "vue": "catalog:"
    },
    "peerDependencies": {
        "vue": ">=3.0.0"
    },
    "files": [
        "dist",
        "LICENSE"
    ],
    "types": "./dist/index.d.ts",
    "main": "./dist/index.js",
    "module": "./dist/index.cjs",
    "exports": {
        ".": {
            "types": "./dist/index.d.ts",
            "import": "./dist/index.js",
            "require": "./dist/index.cjs"
        },
        "./*": "./*"
    },
    "funding": {
        "type": "github",
        "url": "https://github.com/sponsors/alexgrozav"
    },
    "bugs": {
        "url": "https://github.com/inkline/inkline/issues"
    },
    "homepage": "https://inkline.io",
    "keywords": [
        "inkline",
        "ui",
        "ux",
        "dx",
        "library",
        "framework",
        "typescript"
    ],
    "engines": {
        "node": "*",
        "pnpm": "*"
    },
    "license": "ISC",
    "repository": {
        "type": "git",
        "url": "git+https://github.com/inkline/inkline.git"
    }
}
