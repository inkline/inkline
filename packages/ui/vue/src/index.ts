export * from './defaults';
export * from './plugin';
