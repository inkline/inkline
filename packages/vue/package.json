{
    "name": "@inkline/vue",
    "description": "Inkline is the intuitive UI Components library that gives you a developer-friendly foundation for building high-quality, accessible, and customizable Vue.js 3 Design Systems.",
    "version": "1.0.0",
    "type": "module",
    "scripts": {
        "dev": "devtools vite watch",
        "build": "devtools vite build",
        "lint": "devtools eslint",
        "test": "devtools test",
        "test:dev": "devtools test --watch",
        "test:ci": "devtools test --coverage"
    },
    "devDependencies": {
        "@floating-ui/core": "^1.6.2",
        "@floating-ui/dom": "^1.6.5",
        "@inkline/config": "workspace:*",
        "@inkline/core": "workspace:*",
        "@inkline/devtools": "workspace:*",
        "@inkline/eslint-config": "workspace:*",
        "@inkline/plugin": "workspace:*",
        "@inkline/theme": "workspace:*",
        "@inkline/utils": "workspace:*",
        "@testing-library/jest-dom": "catalog:",
        "@testing-library/vue": "catalog:",
        "@vitest/coverage-v8": "catalog:",
        "@vue/runtime-core": "catalog:",
        "eslint": "catalog:",
        "micromustache": "^8.0.3",
        "miragejs": "0.2.0-alpha.3",
        "normalize.css": "^8.0.1",
        "typescript": "catalog:",
        "vite": "catalog:",
        "vite-plugin-dts": "catalog:",
        "vitest": "catalog:",
        "vue": "catalog:"
    },
    "peerDependencies": {
        "@inkline/config": "workspace:*",
        "@inkline/core": "workspace:*",
        "@inkline/theme": "workspace:*",
        "vue": ">=3.0.0"
    },
    "files": [
        "dist",
        "LICENSE",
        "README.md"
    ],
    "main": "./dist/vue.umd.cjs",
    "module": "./dist/vue.js",
    "types": "./dist/vue.d.ts",
    "jsdelivr": "dist/vue.umd.js",
    "unpkg": "dist/vue.umd.js",
    "exports": {
        ".": {
            "import": "./dist/vue.js",
            "require": "./dist/vue.umd.cjs",
            "types": "./dist/vue.d.ts"
        },
        "./*": "./*"
    },
    "funding": {
        "type": "github",
        "url": "https://github.com/sponsors/alexgrozav"
    },
    "bugs": {
        "url": "https://github.com/inkline/inkline/issues"
    },
    "homepage": "https://inkline.io",
    "keywords": [
        "inkline",
        "ui",
        "ux",
        "dx",
        "library",
        "framework",
        "typescript"
    ],
    "engines": {
        "node": "*",
        "pnpm": "*"
    },
    "license": "ISC",
    "repository": {
        "type": "git",
        "url": "git+https://github.com/inkline/inkline.git"
    }
}
