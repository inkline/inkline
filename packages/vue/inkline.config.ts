import { defineConfig } from '@inkline/config';
import { useConfig } from '@inkline/core';
import { useTheme } from '@inkline/theme';
import { useThemeComponents } from './src/theme';

export default defineConfig(() => {
    useTheme();
    useThemeComponents();

    return useConfig();
});
