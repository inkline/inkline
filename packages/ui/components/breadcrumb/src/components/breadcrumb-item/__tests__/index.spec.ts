import { render } from '@testing-library/vue';
import { BreadcrumbItem } from '../index';

import { createTestingInklineOptionsProvide } from '@inkline/test-utils';

const stubs = {
    Linkable: true
};

describe('Components', () => {
    describe('BreadcrumbItem', () => {
        it('should be named correctly', () => {
            expect(BreadcrumbItem.name).toEqual('BreadcrumbItem');
        });

        it('should render correctly', () => {
            const wrapper = render(BreadcrumbItem, {
                global: {
                    stubs,
                    provide: {
                        ...createTestingInklineOptionsProvide()
                    }
                }
            });

            expect(wrapper.html()).toMatchSnapshot();
        });

        describe('computed', () => {
            describe('classes', () => {
                it('should add classes based on props', () => {
                    const wrapper = render(BreadcrumbItem, {
                        props: {
                            active: true,
                            disabled: true
                        },
                        global: {
                            stubs,
                            provide: {
                                ...createTestingInklineOptionsProvide()
                            }
                        }
                    });

                    expect(wrapper.container.firstChild).toHaveClass('-active', '-disabled');
                });
            });

            describe('tabIndex', () => {
                it('should be -1 if disabled', () => {
                    const wrapper = render(BreadcrumbItem, {
                        props: {
                            disabled: true
                        },
                        global: {
                            stubs,
                            provide: {
                                ...createTestingInklineOptionsProvide()
                            }
                        }
                    });
                    const link = wrapper.container.querySelector('linkable-stub');

                    expect(link).toHaveAttribute('tabindex', '-1');
                });

                it('should be -1 if active', () => {
                    const wrapper = render(BreadcrumbItem, {
                        props: {
                            active: true
                        },
                        global: {
                            stubs,
                            provide: {
                                ...createTestingInklineOptionsProvide()
                            }
                        }
                    });
                    const link = wrapper.container.querySelector('linkable-stub');

                    expect(link).toHaveAttribute('tabindex', '-1');
                });

                it('should be 0 otherwise', () => {
                    const wrapper = render(BreadcrumbItem, {
                        global: {
                            stubs,
                            provide: {
                                ...createTestingInklineOptionsProvide()
                            }
                        }
                    });
                    const link = wrapper.container.querySelector('linkable-stub');

                    expect(link).toHaveAttribute('tabindex', '0');
                });
            });

            describe('aria-current', () => {
                it('should be location if active', () => {
                    const wrapper = render(BreadcrumbItem, {
                        props: {
                            active: true
                        },
                        global: {
                            stubs,
                            provide: {
                                ...createTestingInklineOptionsProvide()
                            }
                        }
                    });
                    const link = wrapper.container.querySelector('linkable-stub');

                    expect(link).toHaveAttribute('aria-current', 'location');
                });
            });
        });
    });
});
