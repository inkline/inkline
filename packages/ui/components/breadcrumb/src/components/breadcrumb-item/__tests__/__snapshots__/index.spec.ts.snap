// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components > BreadcrumbItem > should render correctly 1`] = `
"<li class="breadcrumb-item" role="menuitem">
  <linkable-stub active="false" tag="a" tabindex="0" aria-disabled="false" aria-current="false"></linkable-stub>
</li>"
`;
